File: Client_Root/Client/Assets/Scripts/Network/Messages/ToClient/JoinLobbyToC.cs
using FlatBuffers;

public class JoinLobbyToC : IMessage
{
    public const ushort MESSAGE_ID = MessageID.JoinLobbyToC_ID;

    public int m_nResult;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        FBS.JoinLobbyToC.StartJoinLobbyToC(m_Builder);
        FBS.JoinLobbyToC.AddResult(m_Builder, m_nResult);
        var data = FBS.JoinLobbyToC.EndJoinLobbyToC(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.JoinLobbyToC.GetRootAsJoinLobbyToC(buf);

        m_nResult = data.Result;

        return true;
    }
}

File: Server_Root/Server/Server/Messages/ToClient/JoinLobbyToC.cpp
#include "stdafx.h"
#include "JoinLobbyToC.h"
#include "../../../FBSFiles/JoinLobbyToC_generated.h"

JoinLobbyToC::JoinLobbyToC()
{
}

JoinLobbyToC::~JoinLobbyToC()
{
}

unsigned short JoinLobbyToC::GetID()
{
	return MESSAGE_ID;
}

IMessage* JoinLobbyToC::Clone()
{
	return NULL;
}

const char* JoinLobbyToC::Serialize(int* pLength)
{
	FBS::JoinLobbyToCBuilder data_builder(m_Builder);
	data_builder.add_Result(m_nResult);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool JoinLobbyToC::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::JoinLobbyToC>((const void*)pChar);

	m_nResult = data->Result();

	return true;
}